### PR TITLE
CDAP-15861 Google Analytics 360 (SaaS source) Streaming source

### PR DESCRIPTION
**Pull request by anton-kai**

### CODE REVIEW

File: docs/GoogleAnalyticsStreamingSource-streamingsource.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+----------
+### General
+
+**Reference Name:** Name used to uniquely identify this source for lineage, annotating metadata, etc.
+
+**Authorization token:** Authorization token to be used to authenticate in Google Analytics Reporting API.
+
+**Google Analytics View:** The Google Analytics view ID from which to retrieve data.
+
+### Basic
```
Comment (aonischuk) [inline docs/GoogleAnalyticsStreamingSource-streamingsource.md]: Why do we have both "General" and "Basic" sections. These two words have a pretty close meaning. Can we maybe structure configs into sections in a better way?

File: src/main/java/io/cdap/plugin/ga360/source/streaming/GoogleAnalyticsStreamingSource.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+  private GoogleAnalyticsStreamingSourceConfig config;
+
+  public GoogleAnalyticsStreamingSource(GoogleAnalyticsStreamingSourceConfig config) {
+    this.config = config;
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    IdUtils.validateId(config.referenceName);
```
[the remark this answers is not in the document]
Comment (anton-kai) [reply src/main/java/io/cdap/plugin/ga360/source/streaming/GoogleAnalyticsStreamingSource.java]: IdUtils doesn't contain method validateReferenceName

File: src/main/java/io/cdap/plugin/ga360/source/streaming/GoogleAnalyticsStreamingSource.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+/**
+ * Plugin reads data from Google Analytics Reporting API.
+ */
+@Plugin(type = StreamingSource.PLUGIN_TYPE)
+@Name(GoogleAnalyticsStreamingSource.NAME)
+@Description(GoogleAnalyticsStreamingSource.DESCRIPTION)
+public class GoogleAnalyticsStreamingSource extends StreamingSource<StructuredRecord> {
+
+  static final String NAME = "GoogleAnalyticsStreamingSource";
+  static final String DESCRIPTION = "Read data from Google Analytics Reporting API.";
```
[the remark this answers is not in the document]
Comment (anton-kai) [reply src/main/java/io/cdap/plugin/ga360/source/streaming/GoogleAnalyticsStreamingSource.java]: +

[the remark this answers is not in the document]
Comment (anton-kai) [reply src/main/java/io/cdap/plugin/ga360/source/streaming/GoogleAnalyticsStreamingSource.java]: done

File: docs/GoogleAnalyticsStreamingSource-streamingsource.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+Properties
+----------
+### General
+
+**Reference Name:** Name used to uniquely identify this source for lineage, annotating metadata, etc.
+
+**Authorization token:** Authorization token to be used to authenticate in Google Analytics Reporting API.
+
+**Google Analytics View:** The Google Analytics view ID from which to retrieve data.
```
[the remark this answers is not in the document]
Comment (anton-kai) [reply docs/GoogleAnalyticsStreamingSource-streamingsource.md]: done

File: docs/GoogleAnalyticsStreamingSource-streamingsource.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+-----------
+This plugin used to query Google Analytics Reporting API.
+
+Properties
+----------
+### General
+
+**Reference Name:** Name used to uniquely identify this source for lineage, annotating metadata, etc.
+
+**Authorization token:** Authorization token to be used to authenticate in Google Analytics Reporting API.
```
[the remark this answers is not in the document]
Comment (anton-kai) [reply docs/GoogleAnalyticsStreamingSource-streamingsource.md]: done

File: docs/GoogleAnalyticsStreamingSource-streamingsource.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
+# Google Analytics 360 streaming source
+
+Description
+-----------
+This plugin used to query Google Analytics Reporting API.
```
[the remark this answers is not in the document]
Comment (anton-kai) [reply docs/GoogleAnalyticsStreamingSource-streamingsource.md]: done

File: docs/GoogleAnalyticsStreamingSource-streamingsource.md (hunk 1 above)
Comment (anton-kai) [reply docs/GoogleAnalyticsStreamingSource-streamingsource.md]: Basic -> Data retrieval options

File: src/main/java/io/cdap/plugin/ga360/source/streaming/GoogleAnalyticsStreamingSource.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (aonischuk) [reply src/main/java/io/cdap/plugin/ga360/source/streaming/GoogleAnalyticsStreamingSource.java]: use 2.3.0-SNAPSHOT

File: docs/GoogleAnalyticsStreamingSource-streamingsource.md (hunk 1 above)
Comment (aonischuk) [reply docs/GoogleAnalyticsStreamingSource-streamingsource.md]: Possibly yes

File: src/main/java/io/cdap/plugin/ga360/source/streaming/GoogleAnalyticsStreamingSource.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (anton-kai) [reply src/main/java/io/cdap/plugin/ga360/source/streaming/GoogleAnalyticsStreamingSource.java]: done

File: pom.xml [hunk 7]
```diff
@@ -275,6 +275,11 @@
       <artifactId>google-http-client-gson</artifactId>
       <version>1.32.1</version>
     </dependency>
+    <dependency>
+      <groupId>com.google.apis</groupId>
+      <artifactId>google-api-services-analytics</artifactId>
+      <version>v3-rev161-1.25.0</version>
```
Comment (aonischuk) [inline pom.xml]: Can we move the versions to properties? So it's like in other plugins.

Comment (KayAnton) [reply pom.xml]: moved

**Pull request merged**